src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Converters;
+
+namespace System.Text.Json.Node
+{
+    /// <summary>
+    ///   Represents a mutable JSON object.
+    /// </summary>
+    [DebuggerDisplay("JsonObject[{Dictionary.Count}]")]
+    [DebuggerTypeProxy(typeof(DebugView))]
+    public sealed partial class JsonObject : JsonNode
+    {
+        private JsonElement? _jsonElement;
+        private IDictionary<string, JsonNode?>? _dictionary;
```

Comment:
Is there a reason why this is not using a `Dictionary<string, JsonNode?>` directly?
`JsonArray` is using a `List<JsonNode?>`  https://github.com/dotnet/runtime/pull/51025/files#diff-e7e94002d7a28a31460df75e87474da2bbea3d91bb90bc873da332bb704dd3cfR19

Reply:
After this PR is in, the direct use of `Dictionary<,>` will be replaced with a dictionary that has deterministic ordering semantics (so insert\delete produce JSON in an expected manner) but which will still implement `IDictionary<,>`.